pkg/cli/login/login.go

```diff
@@ -89,6 +89,12 @@ func NewCmdLogin(f kcmdutil.Factory, streams genericiooptions.IOStreams) *cobra.
 
 	cmds.Flags().BoolVarP(&o.WebLogin, "web", "w", o.WebLogin, "Login with web browser. Starts a local HTTP callback server to perform the OAuth2 Authorization Code Grant flow. Use with caution on multi-user systems, as the server's port will be open to all users.")
 	cmds.Flags().Int32VarP(&o.CallbackPort, "callback-port", "c", o.CallbackPort, "Port for the callback server when using --web. Defaults to a random open port")
+
+	cmds.Flags().StringVar(&o.OIDCAuthType, "external-auth-type", o.OIDCAuthType, "Specify the authentication type for external issuers to choose which plugin will be used for authentication. Valid values are oidc and azure.")
```

Comment:
It would be a good idea to mark these flags as experimental for 4.15 and even in 4.16